README.md

```diff
@@ -1,23 +1,78 @@
+uppaix6drfps        mongo_controller    replicated          1/1                 martel/mongo-replica-ctrl:latest   
+```  
+
+You can also check the operations performed by the *controller* reading the logs:
+
+```
+$ docker service logs -f mongo_controller
+mongo_controller.1.7x6ujhg5naw7@swarm-1    | INFO:__main__:Waiting mongo service (and tasks) (mongo_mongo) to start
+mongo_controller.1.7x6ujhg5naw7@swarm-1    | ERROR:__main__:Expired attempts waiting for mongo service (mongo_mongo)
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Waiting mongo service (and tasks) (mongo_mongo) to start
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Mongo service is up and running
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:To remove: {'10.0.0.3', '10.0.0.4'}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:To add: {'10.0.0.7', '10.0.0.6'}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:new replSetReconfig: {'ok': 1.0}
+mongo_controller.1.sv8eztwisitz@swarm-manager    | INFO:__main__:Primary is: 10.0.0.6
+```  
+
+To remove the service:
+
+* `docker stack undeploy STACK_NAME`
+* `docker network rm backend`
+
+You can configure the following environment variables for deploying your stack using the provided [`docker-compose.yml`](docker-compose.yml) file (the variables are used in the controller service, so they ar important, without configuring them, the service won't work correctly):
+
+* `MONGO_VERSION`, the default value is `3.2`
+* `REPLICASET_NAME`, the default value is `rs`
+* `MONGO_PORT`, the default value is `27017`
+* `OVERLAY_NETWORK_NAME`, the default value is `backend`
+* `STACK_NAME`, the default value is `mongo`
+* `MONGO_SERVICE_NAME`, the default value is `${STACK_NAME:}_mongo`
+
+
+Few hints, to customize the [`docker-compose.yml`](docker-compose.yml) orchestration according to your needs:
+
+* To use data persistence (which we recommend in production settings), the *Mongo* service needs to be deployed in **global mode**. This is to avoid that more than one instance is deployed on the same node and that different instances concurrently access the same MongoDB data space on the filesystem.
+
+* The *Controller* that maintain the status of the replica-set must be deployed in a single instance over a Swarm manager node. **Multiple instances of the Controller, may perform conflicting actions!** You should ensure that the controller restart in case of error.
+
+* For HA purposes in a production environment your Swarm cluster should have more than one manager. This allows the *Controller* to be start on different nodes in case of issues.
+
+* The `docker-compose.yml` make use of an external network since it is meant to be used in combination with other services that access the Mongo replica-set. To secure the access to the Mongo cluster, you can also comment the `Ports` section in the `docker-compose.yml` file.
```

Comment:
typo: makes

Reply:
We could remove that port mapping by default, couldn't we?

Reply:
Indeed, it is only needed to access the "remotely" the mongo (e.g. for testing purposes). But for now let's keep it.